pkg/auth/dependency/interaction/flows/auth_api.go

```diff
@@ -195,10 +195,37 @@ func (f *AuthAPIFlow) AddLoginID(
 	s, err := f.Interactions.GetInteractionState(i)
 	if err != nil {
 		return err
-	} else if s.CurrentStep().Step != interaction.StepCommit {
-		// in auth api, only password authenticator is supported for login id
-		// password authenticator should be setup during sign up
-		// so the current step must be commit
+	}
+
+	// in auth api, only password authenticator is supported for login id
+	// if the step is StepSetupPrimaryAuthenticator, it must be password authenticator
+	// if user has password authenticator already, the step should be commit
+	ss := s.CurrentStep()
+	if ss.Step == interaction.StepSetupPrimaryAuthenticator &&
+		len(ss.AvailableAuthenticators) > 0 &&
+		ss.AvailableAuthenticators[0].Type == authn.AuthenticatorTypePassword {
+		passwordAuthenticator := ss.AvailableAuthenticators[0]
+
+		// set empty password
```

Comment:
We should clarify that intention is not 'empty' password, but 'no'/'unusable' password such that this cannot be used to authenticate successfully.